helix-term/src/application.rs

```diff
@@ -1029,6 +1037,62 @@ impl Application {
+                                        params.uri
+                                    );
+                                    self.editor.set_error(err);
+                                    Ok(json!(lsp::ShowDocumentResult { success: false }))
+                                }
+                                Ok(path) => {
+                                    match self.editor.open(&path, Action::Replace) {
+                                        Err(err) => {
+                                            let err = format!(
+                                                "failed to open path: {:?}: {:?}",
+                                                params.uri, err
+                                            );
+                                            self.editor.set_error(err);
+                                            Ok(json!(lsp::ShowDocumentResult { success: false }))
+                                        }
+                                        Ok(_) => {
+                                            if let Some(range) = params.selection {
+                                                let (view, doc) = current!(self.editor);
+                                                // TODO: convert inside server
+                                                if let Some(new_range) = lsp_range_to_range(
+                                                    doc.text(),
+                                                    range,
+                                                    offset_encoding,
+                                                ) {
+                                                    let jump =
+                                                        (doc.id(), doc.selection(view.id).clone());
+                                                    view.jumps.push(jump);
+                                                    doc.set_selection(
+                                                        view.id,
+                                                        Selection::single(
+                                                            new_range.anchor,
+                                                            new_range.head,
+                                                        ),
+                                                    );
+                                                    align_view(doc, view, Align::Center);
+                                                }
+                                            }
+
+                                            Ok(json!(lsp::ShowDocumentResult { success: true }))
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
                 };
 
                 let language_server = match self.editor.language_servers.get_by_id(server_id) {
```

Comment:
This can be changed to an unwrap now that we return early if there isn't a language server for this ID above